src/schema/resolvers.js

```diff
@@ -250,124 +255,223 @@ module.exports = {
+      const tx = {
+        _id: txid,
+        txid,
+        version,
+        type: 'APPROVEVOTE',
+        status: 'PENDING',
+        senderAddress,
+        topicAddress,
+        oracleAddress,
+        optionIdx,
+        token: 'BOT',
+        amount,
+        createdTime: Date.now().toString(),
+      };
+      await DBHelper.insertTransaction(Transactions, tx);
+
       return tx;
     },
 
-    createBet: async (root, data, { db: { Transactions } }) => {
-      const version = data.version;
-      const senderAddress = data.senderAddress;
-      const oracleAddress = data.oracleAddress;
-      const amount = data.amount;
-      const optionIdx = data.optionIdx;
+    finalizeResult: async (root, data, { db: { Transactions } }) => {
+      const {
+        version,
+        oracleAddress,
+        senderAddress,
+      } = data;
 
+      // Send finalizeResult tx
       let txid;
       try {
-        const resp = await fetch('http://localhost:5555/bet', {
-          method: 'POST',
-          body: JSON.stringify({
-            oracleAddress,
-            index: optionIdx,
-            amount,
-            senderAddress,
-          }),
-          headers: { 'Content-Type': 'application/json' },
-        }).then(res => res.json());
-
-        txid = resp.result.txid;
+        const tx = await decentralizedOracle.finalizeResult({
+          contractAddress: oracleAddress,
+          senderAddress,
+        });
+        txid = tx.txid;lt.txid;
       } catch (err) {
-        logger.error(`Error call /approve: ${err.message}`);
+        logger.error(`Error calling /finalizeResult: ${err.message}`);
         throw err;
       }
 
+      // Insert Transaction
       const tx = {
         _id: txid,
-        version,
         txid,
-        type: 'BET',
-        txStatus: 'PENDING',
+        version,
+        type: 'FINALIZERESULT',
+        status: 'PENDING',
         senderAddress,
-        entityId: oracleAddress,
-        optionIdx,
-        token: 'QTUM',
-        amount,
+        oracleAddress,
+        createdTime: Date.now().toString(),
       };
+      await DBHelper.insertTransaction(Transactions, tx);
+
+      return tx;
+    },
 
+    withdraw: async (root, data, { db: { Transactions } }) => {
+      const {
+        version,
+        topicAddress,
+        senderAddress,
+      } = data;
+
+      // Send withdraw tx
+      let txid;
       try {
-        await Transactions.insert(tx);
+        const tx = await topicEvent.withdrawWinnings({
+          contractAddress: topicAddress,
+          senderAddress,
+        });
+        txid = tx.txid;
       } catch (err) {
-        logger.error(`Error insert Transactions: ${err.message}`);
+        logger.error(`Error calling /withdraw: ${err.message}`);
         throw err;
       }
 
+      // Insert Transaction
+      const tx = {
+        _id: txid,
+        txid,
+        version,
+        type: 'WITHDRAW',
+        status: 'PENDING',
+        senderAddress,
```

Comment:
do we need a extra tx receipt parsing for withdraw to determine the amount?

Reply:
if we want to return that as part of the `tx` we would have to do a free `callcontract` call. The tx receipt will always only have these 3 fields: txid, sender, hash160

Reply:
@dwalintukan extra `callcontract` call mk senses. But as amount consists of both qtum and bot amounts, probably we should change all `amount` fields to a json format like `{token: qtum, amount: 10}`

Reply:
@frankobe Sure. Would like to do this in another PR though. There are some UI parts that are blocked by this to use live data. Well actually we need to add the Transaction query, but would want to do that in a different PR.